File: 프로그래머스/unrated/142086. 가장 가까운 같은 글자/가장 가까운 같은 글자.py
def solution(s):
    answer = []
    a = []
    for i in range(len(s)):
        if s[i] not in a:
            answer.append(-1)
        elif s[i] in a:
            answer.append(i - ''.join(a).rfind(s[i]))
        a.append(s[i])   
    return answer

File: 프로그래머스/unrated/142086. 가장 가까운 같은 글자/가장 가까운 같은 글자.js
function solution(s) {
    const answer = [];
    const a = [];

    for (let i = 0; i < s.length; i++) {
        if (!a.includes(s[i])) {
            answer.push(-1);
        } else {
            const lastIndex = a.lastIndexOf(s[i]);
            answer.push(i - lastIndex);
        }
        a.push(s[i]);
    }

    return answer;
}